Add MIX hardfork.

ethernomad opened the pull request:
MIX will hardfork in approx 2 weeks.

karalabe committed:
params, swarm: begin Geth v1.8.18, Swarm v0.3.6 cycle

holiman committed:
cmd/clef: encrypt the master seed on disk (#17704)

* cmd/clef: encrypt master seed of clef

Signed-off-by: YaoZengzeng <yaozengzeng@zju.edu.cn>

* keystore: refactor for external use of encryption

* clef: utilize keystore encryption, check flags correctly

* clef: validate master password

* clef: add json wrapping around encrypted master seed

nonsense committed:
swarm, swarm/storage: lower constants for faster tests (#17876)

* swarm/storage: lower constants for faster tests

* swarm: reduce test size for TestLocalStoreAndRetrieve

* swarm: reduce nodes for dec_inc_node_count

acud committed:
cmd/swarm: speed up tests (#17878)

These minor changes already shaved off around 30s.

fjl committed:
rpc: fix subscription corner case and speed up tests (#17874)

Notifier tracks whether subscription are 'active'. A subscription
becomes active when the subscription ID has been sent to the client. If
the client sends notifications in the request handler before the
subscription becomes active they are dropped. The tests tried to work
around this problem by always waiting 5s before sending the first
notification.

Fix it by buffering notifications until the subscription becomes active.
This speeds up all subscription tests.

Also fix TestSubscriptionMultipleNamespaces to wait for three messages
per subscription instead of six. The test now finishes just after all
notifications have been received and doesn't hit the 30s timeout anymore.

karalabe committed:
internal/ethapi: warn on failed account accesses

karalabe committed:
Merge pull request #17887 from karalabe/warn-failed-account-access

internal/ethapi: warn on failed account accesses

fjl committed:
p2p/simulations: fix a deadlock and clean up adapters (#17891)

This fixes a rare deadlock with the inproc adapter:

- A node is stopped, which acquires Network.lock.
- The protocol code being simulated (swarm/network in my case)
  waits for its goroutines to shut down.
- One of those goroutines calls into the simulation to add a peer,
  which waits for Network.lock.

The fix for the deadlock is really simple, just release the lock
before stopping the simulation node.

Other changes in this PR clean up the exec adapter so it reports
node startup errors better and remove the docker adapter because
it just adds overhead.

In the exec adapter, node information is now posted to a one-shot
server. This avoids log parsing and allows reporting startup
errors to the simulation host.

A small change in package node was needed because simulation
nodes use port zero. Node.{HTTP,WS}Endpoint now return the live
endpoints after startup by checking the TCP listener.

fjl committed:
p2p, p2p/discover: add signed ENR generation (#17753)

This PR adds enode.LocalNode and integrates it into the p2p
subsystem. This new object is the keeper of the local node
record. For now, a new version of the record is produced every
time the client restarts. We'll make it smarter to avoid that in
the future.

There are a couple of other changes in this commit: discovery now
waits for all of its goroutines at shutdown and the p2p server
now closes the node database after discovery has shut down. This
fixes a leveldb crash in tests. p2p server startup is faster
because it doesn't need to wait for the external IP query
anymore.

acud committed:
cmd/swarm: split flags and cli command declarations to the relevant files (#17896)

nolash committed:
cmd/swarm: Smoke test for Swarm Feed (#17892)

nolash committed:
swarm/storage: Add accessCnt for GC (#17845)

zelig committed:
swarm/network/stream: generalise setting of next batch (#17818)

* swarm/network/stream: generalize SetNextBatch and add Server SessionIndex

* swarm/network/stream: fix a typo in comment

* swarm/network/stream: remove live argument from NewSwarmSyncerServer

fjl committed:
rpc: fix client shutdown hang when Close races with Unsubscribe (#17894)

Fixes #17837

holiman committed:
tests: update tests, implement no-pow blocks (#17902)

This commit updates our tests with the latest and greatest from ethereum/tests.
It also contains implementation of NoProof for blockchain tests.

mgrachev committed:
cmd/clef: fix typos in README (#17908)

Commit:
accounts/usbwallet: simplify code using -= operator (#17904)

kurkomisi committed:
p2p: meter peer traffic, emit metered peer events (#17695)

This change extends the peer metrics collection:

- traces the life-cycle of the peers
- meters the peer traffic separately for every peer
- creates event feed for the peer events
- emits the peer events

dimchansky committed:
ethdb, rpc: support building on js/wasm (#17709)

The changes allow building WebAssembly applications which use ethclient.Client.

holiman committed:
core/vm: add shortcuts for trivial exp cases (#16851)

jsvisa committed:
cmd/geth: don't set GOMAXPROCS by default (#17148)

This is no longer needed because Go uses all CPUs
by default. The change allows setting GOMAXPROCS in environment if needed.

Smilenator committed:
core/types: fix comment for func SignatureValues (#17921)

holisticode committed:
metrics: added NewCounterForced (#17919)

nonsense committed:
swarm/tracing: disable stdout logging for opentracing (#17931)

Commit:
swarm: Lightnode mode: disable sync, retrieve, subscription (#17899)

* swarm: Lightnode mode: disable sync, retrieve, subscription

* swarm/network/stream: assign error and check in one line

* swarm: restructured RegistryOption initializing

* swarm: empty commit to retrigger CI build

* swarm/network/stream: Added comments explaining RegistryOptions

simon-jentzsch committed:
EIP-1186 eth_getProof (#17737)

* first impl of eth_getProof

* fixed docu

* added comments and refactored based on comments from holiman

* created structs

* handle errors correctly

* change Value to *hexutil.Big in order to have the same output as parity

* use ProofList as return type

acud committed:
swarm/network: disallow historical retrieval requests (#17936)

wuxiangzhou2010 committed:
core: fix a typo (#17941)

jsvisa committed:
accounts: wallet derivation path comment is mistaken (#17934)

fjl committed:
cmd/bootnode: fix -writeaddress output (#17932)

acud committed:
swarm/api/http: remove ModTime=now for direct and multipart uploads (#17945)

holiman committed:
core: better side-chain importing

holisticode committed:
swarm/network/stream: disambiguate chunk delivery messages (retrieval… (#17920)

* swarm/network/stream: disambiguate chunk delivery messages (retrieval vs syncing)

* swarm/network/stream: addressed PR comments

* swarm/network/stream: stream protocol version change due to new message types in this PR

jsvisa committed:
eth/downloader: fix comment typos (#17956)

holiman committed:
core/vm: adds refund as part of the json standard trace (#17910)

This adds the global accumulated refund counter to the standard
json output as a numeric json value. Previously this was not very
interesting since it was not used much, but with the new sstore
gas changes the value is a lot more interesting from a consensus
investigation perspective.

ethernomad committed:
Add MIX hardfork.

fjl committed:
core/state: simplify proof methods (#17965)

This fixes the import cycle build error in core/vm tests.
There is no need to refer to core/vm for a type definition.

jsvisa committed:
eth/downloader: SetBlocksIdle is not used (#17962)

   __
 <(o )___
  ( ._> /
   `---'

whilei reviewed (approved):
LGTM.

Just citing for reference the preceding "half" of these config changes (flag, genesis, bootnodes...) made here: https://github.com/ethoxy/multi-geth/commit/0fa476226c1b7e88de7038c2068e7ad390314e78

johnsBeharry committed:
cmd/clef: replace password arg with prompt (#17897)

* cmd/clef: replace password arg with prompt (#17829)

Entering passwords on the command line is not secure as it is easy to recover from bash_history or the process table.
1. The clef command addpw was renamed to setpw to better describe the functionality
2. The <password> argument was removed and replaced with an interactive prompt

* cmd/clef: remove undeclared variable

holisticode committed:
p2p accounting (#17951)

* p2p/protocols: introduced protocol accounting

* p2p/protocols: added TestExchange simulation

* p2p/protocols: add accounting simulation

* p2p/protocols: remove unnecessary tests

* p2p/protocols: comments for accounting simulation

* p2p/protocols: addressed PR comments

* p2p/protocols: finalized accounting implementation

* p2p/protocols: removed unused code

* p2p/protocols: addressed @nonsense PR comments

holiman committed:
core/vm: check empty in extcodehash

ethernomad committed:
Fix MIX Newark enode.

frncmx committed:
swarm: clean up unused private types and functions (#17989)

* swarm: clean up unused private types and functions

Those that were identified by code inspection tool.

* swarm/storage: move/add Proximity GoDoc from deleted private function

The mentioned proximity() private function was deleted in:
1ca8fc1e6fa0ab4ab1aaca06d6fb32e173cd5f2f

rociiu committed:
cmd/swarm/swarm-smoke: fix issue that loop variable capture in func (#17992)

acud committed:
all: updated code owners file (#17987)

acud committed:
cmd/swarm: auto resolve default path according to env flag (#17960)

KimMachineGun committed:
swarm: modify context key (#17925)

* swarm: modify context key

* gofmt sctx.go

nonsense committed:
swarm/api: unexport Respond methods (#18037)

holisticode committed:
swarm: Better syncing and retrieval option definition (#17986)

* swarm: Better syncing and retrieval option definition

* swarm/network/stream: better comments

* swarm/network/stream: addressed PR comments

rjl493456442 committed:
miner: fix miner stress test (#18039)

SamuelMarks committed:
travis, appveyor: bump to Go 1.11.2 (#18031)

CoreyLin committed:
p2p/protocols: use keyed fields for struct instantiation (#18017)

holiman committed:
downloader: measure successfull deliveries, not failed (#17983)

* downloader: measure successfull deliveries, not failed

* downloader: fix typos

holiman committed:
core, eth/downloader: fix validation flaw, fix downloader printout flaw (#17974)

holiman committed:
signer: remove ineffectual assignments (#18049)

* signer: remove ineffectual assignments

* signer: remove ineffectual assignments

jsvisa committed:
eth/downloader: remove the expired id directly (#17963)

jpeletier committed:
swarm/api: Fix #18007, missing signature should return HTTP 400 (#18008)

fjl committed:
eth/downloader: speed up tests by generating chain only once (#17916)

* core: speed up GenerateChain

Use a mock implementation of ChainReader instead of creating
and destroying a BlockChain object for each generated block.

* eth/downloader: speed up tests by generating chain only once

This change reworks the downloader tests so they share a common test
blockchain instead of generating a chain in every test. The tests are
roughly twice as fast now.

vmaark committed:
swarm/network: light nodes are not dialed, saved and requested from (#17975)

* RequestFromPeers does not use peers marked as lightnode

* fix warning about variable name

* write tests for RequestFromPeers

* lightnodes should be omitted from the addressbook

* resolve pr comments regarding logging, formatting and comments

* resolve pr comments regarding comments and added a missing newline

* add assertions to check peers in live connections

nonsense committed:
swarm, cmd/swarm: address ineffectual assignments (#18048)

* swarm, cmd/swarm: address ineffectual assignments

* swarm/network: remove unused vars from testHandshake

* swarm/storage/feed: revert cursor changes

kurkomisi committed:
p2p: use enode.ID type in metered connection (#17933)

Change the type of the metered connection's id field from string to enode.ID.

CoreyLin committed:
event, event/filter: minor code cleanup (#18061)

LiangMa committed:
p2p: fix comment typo (#18027)

CoreyLin committed:
eth, p2p: fix comment typos (#18014)

CoreyLin committed:
core/vm, eth/tracers: use pointer receiver for GetRefund (#18018)

JoranHonig committed:
common/compiler: capture runtime code and source maps (#18020)

ryanschneider committed:
internal/web3ext: add eth.getProof (#18052)

karalabe committed:
Merge pull request #17982 from holiman/polish_contantinople_extcodehash

core/vm: check empty in extcodehash

fjl committed:
event/filter: delete unused package (#18063)

Commit:
ethclient: include block hash from FilterQuery (#17996)

ethereum/go-ethereum#16734 introduced BlockHash to the FilterQuery
struct. However, ethclient was not updated to include BlockHash in the actual
RPC request.

rjl493456442 committed:
consensus: extend getWork API with block number (#18038)

fjl committed:
core/state: remove lock (#18065)

The lock in StateDB is useless. It's only held in Copy, but Copy is safe
for concurrent use because all it does is read.

kurkomisi committed:
metrics, p2p: add ephemeral registry (#18067)

* metrics, p2p: add ephemeral registry

* metrics: fix linter issue

CoreyLin committed:
rawdb: remove unused parameter for WritePreimages func (#18059)

* rawdb: remove unused parameter for WritePreimages func and modify a
spelling mistake

* rawdb: update the doc for function WritePreimages

randomshinichi committed:
Imply that SwarmApiFlag is the API endpoint to connect to, not to listen on (#18071)

holiman committed:
downloader: different sync strategy

nolash committed:
p2p/simulations, swarm/network: Custom services in snapshot (#17991)

* p2p/simulations: Add custom services to simnodes + remove sim down conn objs

* p2p/simulation, swarm/network: Add selective services to discovery sim

* p2p/simulations, swarm/network: Remove useless comments

* p2p/simulations, swarm/network: Clean up mess from rebase

* p2p/simulation: Add sleep to prevent connect flakiness in http test

* p2p/simulations: added concurrent goroutines to prevent sleeps on simulation connect/disconnect

* p2p/simulations, swarm/network/simulations: address pr comments

* reinstated dummy service

* fixed http snapshot test

rjl493456442 committed:
core: fix default trie cache limit (#17860)

karalabe committed:
cmd, core, eth, light, trie: add trie read caching layer

frncmx committed:
swarm/storage: fix access count on dbstore after cache hit (#17978)

Access count was not incremented when chunk was retrieved
from cache. So the garbage collector might have deleted the most
frequently accessed chunk from disk.

Co-authored-by: Ferenc Szabo <ferenc.szabo@ethereum.org>

mrFranklin committed:
github: format code owners file (#18090)

replace tabs by spaces in the code owners file

mrFranklin committed:
eth: fix minor grammar issue in comment (#18091)

frncmx committed:
swarm/storage: fix garbage collector index skew (#18080)

On file access LDBStore's tryAccessIdx() function created a faulty
GC Index Data entry, because not indexing the ikey correctly.
That caused the chunk addresses/hashes to start with '00' and the last
two digits were dropped. => Incorrect chunk address.

Besides the fix, the commit also contains a schema change which will
run the CleanGCIndex() function to clean the GC index from erroneous
entries.

Note: CleanGCIndex() rebuilds the index from scratch which can take
a really-really long time with a huge DB (possibly an hour).

karalabe committed:
params: update CHTs

karalabe committed:
Merge pull request #18097 from karalabe/update-chts-2

params: update CHTs

nizsheanez committed:
cmd/swarm, swarm/api/http, swarm/bmt, swarm/fuse, swarm/network/stream, swarm/storage, swarm/storage/encryption, swarm/testutil: use pseudo-random instead of crypto-random for test files content generation (#18083)

- Replace "crypto/rand" to "math/rand" for files content generation
- Remove swarm/network_test.go.Shuffle and swarm/btm/btm_test.go.Shuffle - because go1.9 support dropped (see https://github.com/ethereum/go-ethereum/pull/17807 and comments to swarm/network_test.go.Shuffle)

karalabe committed:
params, swarm: release Geth v1.8.18 and Swarm v0.3.6

karalabe committed:
params, swarm: begin Geth v1.8.19 and Swarm v0.3.7 cycle

Commit:
rpc: fix example typo (#18100)

whishes --> wishes

Commit:
rpc: fix package doc typo (#18101)

Changed "send" to "send," in two places

mrFranklin committed:
consensus/clique: fix comment typo (#18103)

Commit:
les: fix pubkey index typo (#18093)

Commit:
internal: fix typo in comments (#18106)

Changed "signTransactions" to "signTransaction"

karalabe committed:
Merge pull request #18087 from karalabe/trie-read-cacher

cmd, core, eth, light, trie: add trie read caching layer

nolash committed:
swarm/storage: Batched database migration (#18113)

mrFranklin committed:
eth: fix comment typo (#18114)

* consensus/clique: fix comment typo

* eth,eth/downloader: fix comment typo

janos committed:
swarm/network/simulation: use simulations.Event instead p2p.PeerEvent (#18098)

holisticode committed:
Swarm accounting (#18050)

* swarm: completed 1st phase of swap accounting

* swarm: swap accounting for swarm with p2p accounting

* swarm/swap: addressed PR comments

* swarm/swap: ignore ErrNotFound on stateStore.Get()

* swarm/swap: GetPeerBalance test; add TODO for chequebook API check

* swarm/network/stream: fix NewRegistry calls with new arguments

* swarm/swap: address @justelad's PR comments

crackcomm committed:
trie: return hasher to pool (#18116)

* trie: return hasher to pool

* trie: minor code formatting fix

karalabe committed:
core, eth/downloader: fix ancestor lookup for fast sync

karalabe committed:
Merge pull request #18085 from holiman/downloader_span

downloader: different sync strategy

karalabe committed:
trie: go fmt package

acud committed:
swarm/api/http: change request served msg log level (#18127)

nonsense committed:
swarm/storage: speed up garbage collection and rpc tests (#18128)

nonsense committed:
cmd/swarm: speed up tests - use global cluster (#18129)

nonsense committed:
cmd/swarm/swarm-smoke: update smoke tests to fit the new scheme for the k8s cluster (#18104)

jpeletier committed:
internal/cmdtest: Expose process exit status and errors (#18046)

frncmx committed:
swarm/storage: move 'running migrations for' log line (#18120)

So that we only see the log message when we actually have to migrate.

nonsense committed:
whisper: log errors on failed tests (#18134)

Debug traces to investigate a travis issue on MacOS

karalabe committed:
core: polish side chain importer a bit

gballet committed:
whisper/mailserver: reduce the max number of opened files (#18142)

This should reduce the occurences of travis failures on MacOS

Also fix some linter warnings

a-sklyarov committed:
Update minimum required Go version in README.md (#18151)

mrFranklin committed:
core: fix comment typo (#18144)

nonsense committed:
swarm/state: refactor InmemoryStore (#18143)

janos committed:
swarm/network/stream: use swarm/mock/mem as mock global store (#18157)

holiman committed:
core: better printout of receipts in bad block reports (#18156)

* core/blockchain: better printout of receipts in bad block reports

* fix splleing

karalabe committed:
Merge pull request #17973 from holiman/splitter2

core: better side-chain importing

enriquefynn committed:
params: add Constantinople block to AllXYZProtocolChanges (#18162)

* params: Add Constantinople block to AllCliqueProtocolChanges

* params: Add Constantinople block to AllEthashProtocolChanges

holiman committed:
trie: fix overflow in write cache parent tracking (#18165)

trie/database: fix overflow in parent tracking

nonsense committed:
travis: increase open file limits (#18155)

frncmx committed:
cmd/swarm: FUSE do not require --ipcpath (#18112)

- Have `${DataDir}/bzzd.ipc` as IPC path default.
- Respect the `--datadir` flag.
- Keep only the global `--ipcpath` flag and drop the local `--ipcpath` flag
  as flags might overwrite each other. (Note: before global `--ipcpath`
  was ignored even if it was set)

fixes ethersphere#795

fjl committed:
eth: increase timeout in TestBroadcastBlock (#18064)

mrFranklin committed:
core: return error if repair block failed (#18126)

* core: return error if repair block failed

* make error a bit shorter

holiman committed:
config: add constantinople block to testchainconfig

jdkanani committed:
fix mixHash/nonce for parity compatible network (#18166)

karalabe committed:
Merge pull request #18179 from holiman/fix_tests

config: add constantinople block to testchainconfig

zsfelfoldi committed:
les: fix fetcher syncing logic (#18072)

janos committed:
swarm/network/simulation: fix New function for-loop scope (#18161)

acud committed:
swarm/api: improve not found error msg (#18171)

Commit:
light: odrTrie tryUpdate should use update (#18107)

TryUpdate does not call t.trie.TryUpdate(key, value) and calls t.trie.TryDelete
instead. The update operation simply deletes the corresponding entry, though
it could retrieve later by odr. However, it adds further network overhead.

nolash committed:
swarm/pss: Message handler refactor (#18169)

chfast committed:
tests: Add flag to use EVMC for state tests (#18084)

jpeletier committed:
Remove multihash from Swarm bzz:// for Feeds (#18175)

holisticode committed:
Accounting metrics reporter (#18136)

nolash committed:
swarm/network: Correct neighborhood depth (#18066)

jpeletier committed:
cmd/swarm: update should error on manifest mismatch (#18047)

* cmd/swarm: fix ethersphere/go-ethereum#979:

update should error on manifest mistmatch

* cmd/swarm: fixed comments and remove sprintf from log.Info

* cmd/swarm: remove unnecessary comment

janos committed:
swarm: add database abstractions (shed package) (#18183)

holisticode committed:
Increase bzz version (#18184)

* swarm/network/stream/: added stream protocol version match tests

* Increase BZZ version due to streamer version change; version tests

* swarm/network: increased hive and test protocol version

LiangMa committed:
p2p/protocols: fix minor comments typo (#18185)

anothel committed:
p2p/discv5: minor code simplification (#18188)

* Update net.go

more simple

* Update net.go

karalabe committed:
p2p/discv5: gofmt

ethernomad committed:
Merge remote-tracking branch 'upstream/master'

holiman committed:
core: more detailed metrics for block processing (#18119)

karalabe committed:
eth/downloader: fix light client cht binary search issue

karalabe committed:
params: update CHTs for the v1.8.19 release

karalabe committed:
Merge pull request #18197 from karalabe/v1.8.19-chts

params: update CHTs for the v1.8.19 release

karalabe committed:
Merge pull request #18196 from karalabe/downloader-cht-fix

eth/downloader: fix light client cht binary search issue

karalabe committed:
params, swarm: release Geth v1.8.19 and Swarm v0.3.7

ethernomad committed:
Merge commit 'dae82f098570e15d44584f0d7f350713f4774727'
